In [16]:
import pandas as pd
from scipy import stats
import numpy as np

In [17]:
df=pd.read_csv('Downloads\Sales_add.csv')

In [18]:
df

Month      Region      Manager  Sales_before_digital_add(in $)  \
0    Month-1  Region - A  Manager - A                          132921   
1    Month-2  Region - A  Manager - C                          149559   
2    Month-3  Region - B  Manager - A                          146278   
3    Month-4  Region - B  Manager - B                          152167   
4    Month-5  Region - C  Manager - B                          159525   
5    Month-6  Region - A  Manager - B                          137163   
6    Month-7  Region - C  Manager - C                          130625   
7    Month-8  Region - A  Manager - A                          131140   
8    Month-9  Region - B  Manager - C                          171259   
9   Month-10  Region - C  Manager - B                          141956   
10  Month-11  Region - B  Manager - C                          159339   
11  Month-12  Region - A  Manager - A                          178939   
12  Month-13  Region - B  Manager - A                          145062   
13  Month-14  Region - A  Manager - A                          151514   
14  Month-15  Region - C  Manager - A                          147463   
15  Month-16  Region - A  Manager - B                          177195   
16  Month-17  Region - A  Manager - C                          140862   
17  Month-18  Region - C  Manager - B                          167996   
18  Month-19  Region - B  Manager - A                          132135   
19  Month-20  Region - A  Manager - B                          152493   
20  Month-21  Region - B  Manager - A                          147425   
21  Month-22  Region - A  Manager - C                          130263   

    Sales_After_digital_add(in $)  
0                          270390  
1                          223334  
2                          244243  
3                          231808  
4                          258402  
5                          256948  
6                          222106  
7                          230637  
8                          226261  
9                          193735  
10                         203175  
11                         276279  
12                         205180  
13                         253131  
14                         229336  
15                         187305  
16                         234908  
17                         191517  
18                         227040  
19                         212579  
20                         263388  
21                         243020

# Hypothesis Testing to test whether there is an increase in sales after stepping into digital marketing


H0= there is no increase in sales after stepping into digital marketing
i.e,Sales after digital add <=sales before digital add
H1=there is an increase in sales after stepping into digital marketing 
i.e, Sales after digital add> sales before digital add

In [19]:
from scipy import stats as stat

In [20]:
Sales_After_digital_add=df['Sales_After_digital_add(in $)']

In [21]:
Sales_After_digital_add

0     270390
1     223334
2     244243
3     231808
4     258402
5     256948
6     222106
7     230637
8     226261
9     193735
10    203175
11    276279
12    205180
13    253131
14    229336
15    187305
16    234908
17    191517
18    227040
19    212579
20    263388
21    243020
Name: Sales_After_digital_add(in $), dtype: int64

In [22]:
Sales_before_digital_add=df['Sales_before_digital_add(in $)']

In [23]:
Sales_before_digital_add

0     132921
1     149559
2     146278
3     152167
4     159525
5     137163
6     130625
7     131140
8     171259
9     141956
10    159339
11    178939
12    145062
13    151514
14    147463
15    177195
16    140862
17    167996
18    132135
19    152493
20    147425
21    130263
Name: Sales_before_digital_add(in $), dtype: int64

In [24]:
t_stat,p_value=stats.ttest_ind(Sales_After_digital_add,Sales_before_digital_add,alternative='greater')

In [25]:
print('T Statistics value is: ',t_stat, 'P value is: ',p_value)

T Statistics value is:  12.995084451110877 P value is:  1.3071840034523225e-16


Here we assume that the significance level is 0.05  , also here the p-value is 1.3071840034523225e-16
So here the p-value is lesser than the significance level so we reject the Null Hypothesis So there is an increase in sales after stepping into digital marketing


# Chi-Square testing for Independency

H0=there is no dependency between the features Region and Manager
H1=there is dependency between the features Region and Manager

In [26]:
df['Region'].value_counts()

Region - A    10
Region - B     7
Region - C     5
Name: Region, dtype: int64

In [27]:
df['Manager'].value_counts()

Manager - A    9
Manager - B    7
Manager - C    6
Name: Manager, dtype: int64

In [28]:
contingency_table=pd.crosstab(df.Region,df.Manager, margins=True)
contingency_table

Manager     Manager - A  Manager - B  Manager - C  All
Region                                                
Region - A            4            3            3   10
Region - B            4            1            2    7
Region - C            1            3            1    5
All                   9            7            6   22

In [29]:
stats.chi2_contingency(contingency_table)

(3.050566893424036,
 0.962256341757093,
 9,
 array([[ 4.09090909,  3.18181818,  2.72727273, 10.        ],
        [ 2.86363636,  2.22727273,  1.90909091,  7.        ],
        [ 2.04545455,  1.59090909,  1.36363636,  5.        ],
        [ 9.        ,  7.        ,  6.        , 22.        ]]))

Here we assume that the level of significance is 0.05 and also here the p-value is 0.962256341757093
So here the of value is greater than the significance level so accept the null hypothesis
That is there is no dependency between the features Region and Manager they are independent